src/Api/Endpoints/Login.php

```diff
@@ -77,56 +75,49 @@ protected function supportedVersions() : array
      */
     public function handle(ServerRequestInterface $request) : ResponseInterface
     {
-        $requestdata = json_decode((string) $request->getBody(), true);
-        if (!isset($requestdata['username']) || !isset($requestdata['password'])) {
-            return (new \LORIS\Http\Response())
-                ->withBody(
-                    new \LORIS\Http\StringStream(
-                        json_encode(
-                            array('error' => 'Missing username or password')
-                        )
-                    )
-                )
-                ->withStatus(400);
+        if (count($request->getAttribute('pathparts')) !== 1) {
+            return new \LORIS\Http\Response\NotFound();
         }
 
-        $user     = $requestdata['username'];
-        $password = $requestdata['password'];
-
-        $login = $this->getLoginAuthenticator();
-
-        if ($login->passwordAuthenticate($user, $password, false)) {
-            $token = $this->getEncodedToken($user);
-            if (!empty($token)) {
-                return (new \LORIS\Http\Response())
-                    ->withBody(
-                        new \LORIS\Http\StringStream(
-                            json_encode(array('token' => $token))
-                        )
-                    )
-                    ->withHeader("Content-Type", "application/json")
-                    ->withStatus(200);
-            } else {
-                return (new \LORIS\Http\Response())
-                    ->withBody(
-                        new \LORIS\Http\StringStream(
-                            json_encode(
-                                array('error' => 'Unacceptable JWT key')
-                            )
-                        )
-                    )
-                    ->withHeader("Content-Type", "application/json")
-                    ->withStatus(500);
+        switch ($request->getMethod()) {
+        case 'POST':
+            $requestdata = json_decode((string) $request->getBody(), true);
+
+            $user     = $requestdata['username'] ?? null;
+            $password = $requestdata['password'] ?? null;
+
+            if ($user === null || $password === null) {
+                return new \LORIS\Http\Response\BadRequest(
+                    'missing username or password'
```

Comment:
nitpick but might be best to maintain capitalization of original err. message

Reply:
adressed